daemon/k8s/namespaces.go

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
I realize this follows the pattern of #36101, but I wonder if there's a way to keep it out of `./daemon` entirely. It seems like the dependency on `FetchNamespace()` could easily be moved out of `./daemon/cmd`. Would that be a useful cleanup to do?

Reply:
Sure I can take a look at further cleaning this up.

Reply:
WDYT about the cleanup in the last commit? It still feels only half-way there as there's still bunch of logic in daemon/cmd/endpoint.go that probably doesn't belong there. Also the fetching of the pod directly from api-server feels very wrong as it'll eventually end up in the pod table anyway, but I didn't feel like changing this right now.

Reply:
This looks a lot better. I'll read through it more carefully now.

I appreciate there is poor separation of concerns between pkg/endpoint and daemon/cmd, and fixing it all at once is untenable. Thanks for taking the first step!